File: src/main/java/cz/cvut/fel/plichjan/distmesh/inputs/DIntersect.java
package cz.cvut.fel.plichjan.distmesh.inputs;

/**
 * function d=dintersect(d1,d2), d=max(d1,d2);
 */
public class DIntersect implements IDistanceFunction {
    private IDistanceFunction d1, d2;

    //function d=dintersect(d1,d2)
    public DIntersect(IDistanceFunction d1, IDistanceFunction d2) {
        this.d1 = d1;
        this.d2 = d2;
    }

    @Override
    public double call(double x, double y) {
        // d=max(d1,d2);
        return Math.max(d1.call(x, y), d2.call(x, y));
    }
}

File: src/main/kotlin/cz/cvut/fel/plichjan/distmesh/inputs/DIntersect.kt
package cz.cvut.fel.plichjan.distmesh.inputs

import kotlin.math.max

/**
 * function d=dintersect(d1,d2), d=max(d1,d2);
 */
data class DIntersect(val d1: IDistanceFunction, val d2: IDistanceFunction) : IDistanceFunction {
    override fun call(x: Double, y: Double): Double {
        return max(d1.call(x, y), d2.call(x, y))
    }
}
